pkg/cmd/root/dynamic_target_provider.go

```diff
@@ -0,0 +1,116 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package root
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/gardener/gardenctl-v2/pkg/target"
+)
+
+// DynamicTargetProvider is a wrapper that combines the basic
+// file-based TargetProvider with CLI flags, to allow the user
+// to change the target for individual gardenctl commands
+// on-the-fly without changing the file on disk every time.
+//
+// If no CLI flags are given, this functions identical to the
+// regular TargetProvider from NewFilesystemTargetProvider().
+//
+// Otherwise, the flags are used to augment the existing target.
+// In this mode, changing the target via Write() is not allowed,
+// so that commands do not accidentally overwrite the target on
+// disk.
+type DynamicTargetProvider struct {
```

Comment:
DynamicTargetProvider implements `target.TargetProvider` right?
Why have you chosen to have `dynamic_target_provider.go` under the `root` package and not under the `target` package?

Reply:
Because the DTP only makes sense in the context of the CLI flags, so I did not want to introduce a cobra dependency in the `target` package.